src/main/java/tdl/s3/upload/FileUploaderImpl.java

```diff
@@ -31,7 +34,11 @@ public void upload(File file, String path) {
 
     @Override
     public boolean exists(String path) {
-        return destination.canUpload(path);
+        try {
+            return destination.canUpload(path);
+        } catch (DestinationOperationException ex) {
+            return false;
```

Comment:
What do you think of this?

https://github.com/julianghionoiu/s3-sync-stream/pull/26/commits/4042eb3fd8ee1f707bda0bc64014649994ed7884